### DISCUSSION TITLE
How to combine OR and AND operators?

### DISCUSSION THREAD

**Discussion by nkvch** [prisma/prisma#11769] [Q&A] [answered]
Hi, I'm trying to do a nested query like this:

```
prisma.someModel.findMany({
    where: {
      AND: [
        {
          OR: [
            { firstName: { contains: 'John' } },
            { lastName: { contains: 'John' } }
          ]
        },
        {
          OR: [
            { firstName: { contains: 'Doe' } },
            { lastName: { contains: 'Doe' } }
          ]
        }
      ]
    }
})
```

So that it could find both record where
```
{
    firstName: 'John',
    lastName: 'Doe',
}
```
and where
```
{
    firstName: 'Doe',
    lastName: 'John',
}
```

In MySQL it would be sth like this I guess:
```
SELECT * FROM some_model
WHERE (
    firstName like '%John%'
    OR
    lastName like '%John%'
) AND (
    firstName like '%Doe%'
    OR
    lastName like '%Doe%'
)
```

Have tried it in mysql and it worked. Is there a way to obtain sth like this in prisma?

**Comment by nkvch** [comment]
Basically, is there a way to AND two ORs ?

**Comment by Luke265** [comment, marked as answer]
I'm not sure if I'm missing something. But your prisma query basically already generates same SQL query that you wrote below.

**Reply by nkvch** [reply]
Yes, but is doesn't return anything...

**Reply by nkvch** [reply]
Oh, sorry, the problem was idiotic and it was in code :D

**Reply by DanielTrybe** [reply]
how you solve this?

**Reply by thatjsprof** [reply]
@nkvch what was the solution?

**Reply by nkvch** [reply]
@thatjsprof @DanielTrybe 
I simplified the search in the end, so I don't have the code that worked. But, as far as I remember I solved it by simplifying the query to:
```
prisma.someModel.findMany({
    where: {
      OR: [
        {
          firstName: { contains: 'John' },
          lastName: { contains: 'Doe' }
        },
        {
          firstName: { contains: 'Doe' },
          lastName: { contains: 'John' }
        }
      ]
    }
})
```
Which served exactly same purpose as my first overcomplicated initial query

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
